amplify/backend/function/getUserNotificationsHMAC/src/index.js

```diff
@@ -0,0 +1,29 @@
+const Base64 = require('crypto-js/enc-base64');
+const hmacSHA256 = require('crypto-js/hmac-sha256');
+
+let magicbellApiSecret = process.env.MAGICBELL_API_SECRET;
+
+const setEnvVariables = async () => {
+  const ENV = process.env.ENV;
+
+  if (ENV === 'qa' || ENV === 'prod') {
+    const { getParams } = require('/opt/nodejs/getParams');
+    [magicbellApiSecret] = await getParams(['magicbellApiSecret']);
+  }
+};
+
+exports.handler = async (event) => {
+  try {
+    await setEnvVariables();
+  } catch (err) {
+    throw new Error('Unable to set environment variables. Reason:', err);
+  }
+
+  const checksummedWalletAddress = event.request.headers['x-wallet-address'];
```

Comment:
So I still don't think I trust this, as I can just pass the header onto the lambda itself and just get a valid HMAC back without any validation.

If I remember correctly, the `x-wallet-address` header was put in as a util to have access to the address, not necessarily as something secure, because in the end, it's just a string.

I think we might need to involve both @jakubcolony (he set up the header in the first place), @bogdan-1337 (so we know the limits of how we can call the lambdas) as well as any other interested parties and just have a call to figure this out.

But as it stands, I don't think it's secure enough.

Reply:
So as far as I understand it (and after a brief chat with Bogdan to confirm), we cannot call the lambda from anywhere outside of the AWS console / api, and only with our Colony credentials. When we call the graphql query which resolves using the lambda, we call it with an auth token, not passing the wallet address directly, so even if somebody tried to hack that graphql call they wouldn't be able to just add a different wallet address.

Let's indeed have a call tomorrow when Bogdan is available though, we can figure it out together for sure 💪